api/src/main/java/org/infinispan/api/async/AsyncCache.java

```diff
@@ -43,123 +47,409 @@
+    */
+   Flow.Publisher<K> keys(CacheOperationOptions options);
+
+   /**
+    * Retrieve all entries
+    *
+    * @return
+    */
+   default Flow.Publisher<CacheEntry<K, V>> entries() {
+      return entries(CacheOperationOptions.DEFAULT);
+   }
 
    /**
     * Retrieve all entries
     *
+    * @param options
     * @return
     */
-   Flow.Publisher<? extends CacheEntry<K, V>> entries();
+   Flow.Publisher<CacheEntry<K, V>> entries(CacheOperationOptions options);
 
    /**
     * @param entries
     * @return Void
     */
-   CompletionStage<Void> putAll(Map<K, V> entries);
+   default CompletionStage<Void> putAll(Map<K, V> entries) {
+      return putAll(entries, CacheWriteOperationOptions.DEFAULT);
+   }
+
+   /**
+    * @param entries
+    * @param options
+    * @return
+    */
+   CompletionStage<Void> putAll(Map<K, V> entries, CacheWriteOperationOptions options);
 
    /**
     * Retrieves the entries for the specified keys.
     *
     * @param keys
     * @return
     */
-   Flow.Publisher<? extends CacheEntry<K, V>> getMany(Set<K> keys);
+   default Flow.Publisher<CacheEntry<K, V>> getAll(Set<K> keys) {
+      return getAll(keys, CacheOperationOptions.DEFAULT);
+   }
+
+   /**
+    * Retrieves the entries for the specified keys.
+    *
+    * @param keys
+    * @param options
+    * @return
+    */
+   Flow.Publisher<CacheEntry<K, V>> getAll(Set<K> keys, CacheOperationOptions options);
 
    /**
     * Removes a set of keys. Returns the keys that were removed.
     *
     * @param keys
     * @return
     */
-   Flow.Publisher<K> removeAll(Set<K> keys);
+   default Flow.Publisher<K> removeAll(Set<K> keys) {
+      return removeAll(keys, CacheOperationOptions.DEFAULT);
+   }
 
    /**
     * Removes a set of keys. Returns the keys that were removed.
     *
     * @param keys
+    * @param options
+    * @return
+    */
+   Flow.Publisher<K> removeAll(Set<K> keys, CacheOperationOptions options);
+
+   /**
+    * Removes a set of keys. Returns the keys that were removed.
+    *
+    * @param keys
+    * @return
+    */
+   default Flow.Publisher<CacheEntry<K, V>> getAndRemoveAll(Set<K> keys) {
+      return getAndRemoveAll(keys, CacheOperationOptions.DEFAULT);
+   }
+
+   /**
+    * Removes a set of keys. Returns the keys that were removed.
+    *
+    * @param keys
+    * @param options
     * @return
     */
-   Flow.Publisher<? extends CacheEntry<K, V>> getAndRemoveAll(Set<K> keys);
+   Flow.Publisher<CacheEntry<K, V>> getAndRemoveAll(Set<K> keys, CacheOperationOptions options);
 
    /**
     * Estimate the size of the store
     *
     * @return Long, estimated size
     */
-   CompletionStage<Long> estimateSize();
+   default CompletionStage<Long> estimateSize() {
+      return estimateSize(CacheOperationOptions.DEFAULT);
+   }
+
+   /**
+    * Estimate the size of the store
+    *
+    * @param options
+    * @return Long, estimated size
+    */
+   CompletionStage<Long> estimateSize(CacheOperationOptions options);
```

Comment:
we do not longer support accurate size?

Reply:
We decided we didn't want to expose it anymore, at least not directly. We may add it further along